Content.Shared/_Sunrise/Weapons/Ranged/ConsumableAmmoSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Robust.Shared.GameObjects;
+using Content.Shared._Sunrise.Weapons.Ranged;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Stacks;
+using Content.Shared.Popups;
+using Content.Shared.Weapons.Ranged.Events;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using System;
+using Content.Shared.Projectiles;
+using Content.Shared.Interaction;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Examine;
+
+namespace Content.Shared._Sunrise.Weapons.Ranged
+{
+    public sealed class ConsumableAmmoSystem : EntitySystem
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedStackSystem _stackSystem = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly SharedProjectileSystem _projectileSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ConsumableAmmoComponent, InteractUsingEvent>(OnLoadAmmo);
+            SubscribeLocalEvent<ConsumableAmmoComponent, ShotAttemptedEvent>(OnShotAttempted);
+            SubscribeLocalEvent<ConsumableAmmoComponent, TakeAmmoEvent>(OnTakeAmmo);
+            SubscribeLocalEvent<ConsumableAmmoComponent, ExaminedEvent>(OnExamine);
+        }
+
+        private void OnLoadAmmo(EntityUid uid, ConsumableAmmoComponent ammo, InteractUsingEvent args)
```

Comment:
```suggestion
        private void OnLoadAmmo(Entity<ConsumableAmmoComponent> ent, ref InteractUsingEvent args)
```

Используй новый формат подписки через Entity<>

Компонент получаешь как ent.Comp, а для EntityUid просто передаешь ent или ent.Owner, если будет ошибка